### PR TITLE
Maintain HTML provided case for TagHelper attributes

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue #266 
@NTaylorMullen @dougbu

### CODE REVIEW

File: src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs [hunk 1]
```diff
@@ -200,7 +200,9 @@ protected virtual string GenerateUniqueId()
                 if (matchingAttributes.Any())
                 {
                     // First attribute wins, even if there's duplicates.
-                    var attributeValueChunk = matchingAttributes.First().Value;
+                    var firstAttribute = matchingAttributes.First();
+                    var attributeName = firstAttribute.Key;
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs]: Move this down to where it's used.

File: test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Output/TagHelpersInSection.cs [hunk 2]
```diff
@@ -70,7 +70,7 @@ public TagHelpersInSection()
 #line hidden
                 __tagHelperStringValueBuffer = EndTagHelperWritingScope();
                 __MyTagHelper.BoundProperty = __tagHelperStringValueBuffer.ToString();
-                __tagHelperExecutionContext.AddTagHelperAttribute("BoundProperty", __MyTagHelper.BoundProperty);
+                __tagHelperExecutionContext.AddTagHelperAttribute("boundproperty", __MyTagHelper.BoundProperty);
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Output/TagHelpersInSection.cs]: Should do a mix-cased test just to validate.

**Comment by NTaylorMullen** [comment]
:watch:

File: test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Output/TagHelpersInSection.cs (hunk 2 above)
Comment (ajaybhargavb) [inline test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Output/TagHelpersInSection.cs]: Added.

**Comment by ajaybhargavb** [comment]
Updated.

**Comment by dougbu** [comment]
:shipit: if @NTaylorMullen is happy

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request merged**